### ISSUE TITLE
Oversampling with list file and im2rec

### ISSUE THREAD

**Issue by RishabGargeya** [dmlc/mxnet#391]
Hi, I'm trying to created an oversampled list file for a training dataset of  images with a total of size ~25gb. Since my dataset is imbalanced, I'm trying to oversample certain classes by simply repeating rows in the list file pertaining to images of that class. When running im2rec on the over samples list file, I get an output of about ~1.7 Gb. The problem with this approach is that running a neural net on this file doesn't shuffle the list file during training, leading to many sequential batches full of the same class in the order of the list file. Is there something I am missing? Did I create my list file or rec file incorrectly? Thank you very much for your time.

**Comment by tqchen** [comment]
You will need to preshuffle the list before you pack the file.

**Comment by RishabGargeya** [comment]
Thanks for the reply. The minibatch iterations are now properly shuffled.
